app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -140,8 +140,79 @@ def load_balanced?
+      #        "23"=>{"counts"=>{"rpm"=>32, "erratum"=>4}, "metadata"=>{"env_id"=>3, "product_id"=>1, "content_type"=>"yum", "library_instance_id"=>14}}}}}}
+      #
+
+      def repos_content_count(repos)
+        new_content_counts = initialize_content_counts
+        repos.each do |repo|
+          process_repository(repo, new_content_counts)
+        end
+        remove_unavailable_versions(new_content_counts)
+        update(content_counts: new_content_counts)
+      end
+
+      def initialize_content_counts
+        (content_counts.deep_dup || { content_view_versions: {} }).with_indifferent_access
+      end
+
+      def process_repository(repo, content_counts)
+        repo_mirror_service = repo.backend_service(self).with_mirror_adapter
+        repo_content_counts = repo_mirror_service.latest_content_counts
+        translated_counts = translate_counts(repo, repo_mirror_service, repo_content_counts)
+        content_counts[:content_view_versions][repo.content_view_version_id.to_s] ||= { repositories: {}}
+        content_counts[:content_view_versions][repo.content_view_version_id.to_s][:repositories][repo.id.to_s] = translated_counts
+      end
+
+      def translate_counts(repo, repo_mirror_service, repo_content_counts)
+        translated_counts = {metadata: {}, counts: {}}
+        translated_counts[:metadata] = {
+          env_id: repo.environment_id,
+          library_instance_id: repo.library_instance_or_self.id,
+          product_id: repo.product_id,
+          content_type: repo.content_type
+        }
+        repo_content_counts&.each do |name, count|
+          count = count[:count]
+          if name == 'rpm.package' && repo.content_counts['srpm'] > 0
+            translated_counts[:counts]['srpm'] = repo_mirror_service.count_by_pulpcore_type(::Katello::Pulp3::Srpm)
+            translated_counts[:counts]['rpm'] = count - translated_counts[:counts]['srpm']
+          elsif name == 'container.manifest' && repo.content_counts['docker_manifest_list'] > 0
+            translated_counts[:counts]['docker_manifest_list'] = repo_mirror_service.count_by_pulpcore_type(::Katello::Pulp3::DockerManifestList)
+            translated_counts[:counts]['docker_manifest'] = count - translated_counts[:counts]['docker_manifest_list']
+          else
+            translated_counts[:counts][::Katello::Pulp3::PulpContentUnit.katello_name_from_pulpcore_name(name, repo)] = count
+          end
+        end
+        translated_counts
+      end
+
+      def remove_unavailable_versions(content_counts)
+        version_ids_available_to_proxy = Katello::ContentViewVersion.in_environment(lifecycle_environments)&.pluck(:id)&.uniq
+        version_ids_in_count_map = content_counts[:content_view_versions].keys&.map(&:to_i)
+        version_ids_to_remove = version_ids_in_count_map - version_ids_available_to_proxy
+        version_ids_to_remove.each { |id| content_counts[:content_view_versions].delete(id.to_s) }
+      end
+
+      def global_content_counts
```

Comment:
What do you think about refactoring global_content_counts to use `repos_content_count`? It could look like: 

```ruby
def global_content_counts
  new_content_counts = { content_view_versions: {} }
  repos = fetch_repositories # smart_proxy_helper.repositories_available_to_capsule(environment, content_view)
  repos_content_count(repos)
end
```

This way, we wouldn't have to maintain the funky special cases for srpms and manifest lists in two different places.
I don't think the global counting code is doing anything more efficient then the new code you added.

Reply:
In my head, I thought I had already done that.. 😄  Thanks for catching it..Will push..